include/igraph.h

```diff
@@ -99,4 +99,6 @@
 #include "igraph_eulerian.h"
 #include "igraph_graphicality.h"
 
+#include "igraph_cycles.h"
```

Comment:
There isn't really any structure in `igraph.h` so I'd just add this header without an empty line between them and the others. The way I see it, there are three groups of headers in this file: very-basic-lowlevel-stuff, templated types and all the rest. This one belongs in the "all the rest" section.

Reply:
Of course, I actually meant fixed this before asking for a review.

Reply:
How about merging `igraph_eulerian.h` and `igraph_cycles.h`? Both deal with "cycles" in some sense.

Reply:
`igraph_eulerian.h` also deals with Eulerian _paths_, doesn't it? (But we also have `igraph_paths.h` :)).
Let's keep things as is, just move `igraph_cycles.h` together with the rest of the group above.